SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -838,7 +852,7 @@ bool SpatialGDKGenerateSchemaForClasses(TSet<UClass*> Classes, FString SchemaOut
 	FComponentIdGenerator IdGenerator = FComponentIdGenerator(NextAvailableComponentId);
 
 	GenerateSchemaFromClasses(TypeInfos, SchemaOutputPath, IdGenerator);
-	GenerateSchemaForSublevels(SchemaOutputPath, IdGenerator);
+	GenerateSchemaForSublevels(GetLevelNamesToPaths(), SchemaOutputPath);
```

Comment:
This method creates an IdGenerator internally using NextAvailableComponentId, but the GenerateSchemaForClasses doesn't update the global.

We could refactor GenerateSchemaForSublevels to take an IdGenerator parameter again and also provide an overload which doesn't that just does
```
GenerateSchemaForSublevels(GetLevelNamesToPaths(), SchemaOutputPath) {
    FComponentIdGenerator IdGenerator = FComponentIdGenerator(NextAvailableComponentId);
    GenerateSchemaForSublevels(GetLevelNamesToPaths(), SchemaOutputPath, IdGenerator);
    NextAvailableComponentId = IdGenerator.Peek();
}
```